### PR TITLE
Add a Makefile

### PR DESCRIPTION
**Pull request by geoknee**
Also fixes bug where we are not enforcing the javascript codegen in CI (means addresses.json is out of date).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+### Auto-generated files
+codegen-all: go-codegen js-codegen
+
+go-codegen:
+	go run superchain/internal/codegen/main.go
+
+js-codegen:
+	node ./scripts/codegen.js
```
Comment (geoknee) [inline Makefile:8]: This doesn't really need to be split out.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	go test ./superchain/... -v
+
+test-validation:
+	go test ./validation/... -v
+
+### Cleaning
+clean-add-chain:
+	rm -f superchain/configs/sepolia/testchain_*.yaml
+	rm -f superchain/extra/addresses/sepolia/testchain_*.json
+	rm -f superchain/extra/genesis-system-configs/sepolia/testchain_*.json
```
Comment (geoknee) [inline Makefile:34]: Do we still need this?

**Review by geoknee — COMMENTED** [review]
Also need to update the readme to use `make` commands!

How about filtering tests to a particular chain?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+### Testing
+test-all: test-add-chain test-superchain test-validation
+
+test-add-chain:
+# We separate the first test from the rest because it generates artifacts
+# Which need to exist before the remaining tests run.
+	go test ./add-chain/... -run TestAddChain_Main -v
+	go test ./add-chain/... -run '[^TestAddChain_Main]' -v
+	make clean-add-chain
```
[the remark this answers is not in the document]
Comment (bitwiseguy) [reply Makefile:22]: By default the make command will exit with a non-zero exit code (and stop execution) if any one of those commands fails

File: Makefile (hunk 2 above)
Comment (bitwiseguy) [reply Makefile:34]: I think that one is handy to have around

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (geoknee) [reply Makefile:22]: Thanks, will resolve this.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	# go run ./add-chain check-rollup-config
+	mkdir -p ./superchain/extra/genesis/$(SUPERCHAIN_TARGET)
+	cd $(MONOREPO_DIR) && go run ./op-chain-ops/cmd/registry-data \
+		--l2-genesis=$(GENESIS_CONFIG) \
+		--bytecodes-dir=$(ROOT_DIR)/superchain/extra/bytecodes \
+		--output=$(ROOT_DIR)/superchain/extra/genesis/$(SUPERCHAIN_TARGET)/$(CHAIN_SHORT_NAME).json.gz
+	make codegen
+
+promote-to-standard:
+	go run ./add-chain promote-to-standard --chain-id=${chain-id}
```
Comment (bitwiseguy) [inline Makefile:18]: Should the arg be changed to `$(CHAIN_ID)` to read the env var?

File: .circleci/config.yml [hunk 5]
```diff
@@ -24,25 +25,16 @@ jobs:
...
-      - run:
-          name: tidy validation module
-          command: go mod tidy
-          working_directory: validation
-      - run:
-          name: tidy add-chain module
-          command: go mod tidy
-          working_directory: add-chain
+          name: tidy
+          command: make tidy-all
```
Comment (bitwiseguy) [inline .circleci/config.yml:37]: Do you need to do `cp .env.example .env` before running the make commands in CI?

Comment (geoknee) [reply .circleci/config.yml:37]: Very likely, yes!

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+tidy-add-chain:
+	cd add-chain && go mod tidy
+
+tidy-superchain:
+	cd superchain && go mod tidy
+
+tidy-validation:
+	cd validation && go mod tidy
+
+### Removing a chain, example: make remove-chain superchain_target=sepolia chain=mychain
```
Comment (bitwiseguy) [inline Makefile:63]: Instead of requiring two args for this command could you just pass in chain id?

**Comment by geoknee** [comment]
We could also consider using `just` instead of `make` https://github.com/casey/just.

File: Makefile (hunk 4 above)
Comment (geoknee) [reply Makefile:18]: I don't think we have that in the `.env.example` currently (we read it from the rollup.json file during `add-chain`).

**Comment by geoknee** [comment]
In favour of #366

**Pull request closed without merging**